src/gui/clips/AutomationClipView.cpp

```diff
@@ -487,4 +496,58 @@ void AutomationClipView::scaleTimemapToFit( float oldMin, float oldMax )
 }
 
 
+
+
+bool AutomationClipView::splitClip(const TimePos pos)
+{
+	setMarkerEnabled(false);
+
+	const TimePos splitPos = m_initialClipPos + pos;
+
+	// Don't split if we slid off the Clip or if we're on the clip's start/end
+	// Cutting at exactly the start/end position would create a zero length
+	// clip (bad), and a clip the same length as the original one (pointless).
+	if (splitPos <= m_initialClipPos || splitPos >= m_initialClipEnd) { return false; }
+
+	m_clip->getTrack()->addJournalCheckPoint();
+	m_clip->getTrack()->saveJournallingState(false);
+
+	auto rightClip = new AutomationClip(*m_clip);
+	auto leftClip = new AutomationClip(*m_clip);
+
+	rightClip->clear();
+	// Remove the nodes past the split point from the left clip. +1 to prevent deleting a node
+	// right on the split point
+	leftClip->removeNodes(splitPos + 1, m_initialClipEnd);
```

Comment:
For some reason this doesn't seem to be working, since the nodes to the right of the split point still remain in the left clip after the split. It doesn't make a difference though because the left clip's length is shortened which prevents nodes past the clip length from having an effect.

This might be a good thing, because with Cubic Hermite automations, neighboring nodes can affect the shape of the waveform not just between the node and its neighbors but also between its neighbors and *their* neighbors. This means that simply chopping off nodes to the right of the split point in a Cubic Hermite automation clip cannot really be done without changing the shape of the waveform to the left of the split point, even if you add a new node at the split point.

Simply shortening the length of the left automation clip without removing any of the nodes seems to solve that problem for Cubic Hermite clips, but the problem remains for the right clip. Unfortunately the length of automation clips cannot be modified from the left side the way sample clips can - only the right side - so the clip to the right of the split point cannot be fixed in the same way.

I think these are our options:
1. Make automation clip length adjustable from the left side too, similar to sample clips 
2. Pretend the problem with splitting Cubic Hermite clips doesn't exist
3. Return `false` for Cubic Hermite clips (and possibly print a command-line error message) to disallow splitting them

I think option 1 would be difficult and probably beyond the scope of this PR, and I can't think of any other genuine solution. So that leaves options 2 and 3. Personally I favor option 3 to avoid the strong possibility of messing up automations when splitting them.

What are your thoughts?

Reply:
For some reason the nodes to the right are being deleted for me; I believe my branch is up to date, so I'm not sure why it is not working for you.

Regarding splitting Cubic Hermite clips; that was also on my mind, and I agree I don't believe it's impossible to correctly reconstruct the curve by inserting a new point in the middle. Option 3 seems a bit inconvenient for the user, since I think an approximate split is better than no split at all. I was originally planning to let it slide with the current implementation, but now that you bring it up, I do think it would be good to take the time to do it correctly and make sure everything is exact. I think I will try and see how hard it is to do option 1 and implement ResizeLeft for automation clips.

Reply:
Done :) That only took about 30 minutes to code, not bad at all!

I added the ability to resize automation clips from the left (I had to add offsets to a few variables in paintEvent; I think I noticed a minor graphical bug with vertical lines sometimes appearing, I will look into that), and I changed the splitClip code to essentially match SampleClip and PatternClip. This does however mean that when editing the right split clip, all the previous nodes are in front of you, which may be a bit odd, but it works.

(Now I'm thinking; it might be good to also change the timeline numbering in the AutomationEditor to help users see where exactly the clip starts...)